### PR TITLE
Power plant boiler fire side and steam heater models, docs, and tests

### PR DESCRIPTION
**Pull request by MAZamarripa**
**NOTE:** When this PR is approved, you *must* do the following so it can be merged:
* Add the "approved" tag to the pull-request
* Trigger tests by pushing a commit or manually re-running in CircleCI

See [this wiki page](https://github.com/IDAES/idaes-dev/wiki/Github-Pull-Request-Workflow-with-Approval-Tag) for details.

----
## Fixes
contributes models to the power plant library

## Summary/Motivation:
WIP: complete power plant library

## Changes proposed in this PR:
- adding boiler fire side model, test, and documentation
- adding steamheater model, test, and documentation

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

----

## Reviewer Checklist
* Reviewer 1
- [ ] Tests: do they pass, do they cover functionality, are they understandable?
- [ ] Documentation: is the code commented, are there user-level docs where needed?
* Reviewer 2
- [ ] Tests: do they pass, do they cover functionality, are they understandable?
- [ ] Documentation: is the code commented, are there user-level docs where needed?

### CODE REVIEW

File: idaes/power_generation/unit_models/steamheater.py [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+        def radius_out(b):
+            return b.radius_in + b.tube_thickness
+
+        # Thickness of fin
+        self.fin_thickness = Var(
+                initialize=0.004,
+                doc="Thickness of fin")
+
+        # Half of fin thickness
+        @self.Expression(doc="Half of fin thickness")
```
Comment (andrewlee94) [inline idaes/power_generation/unit_models/steamheater.py:231]: I do have to ask if we really need a named `Expression` for this.

File: idaes/power_generation/unit_models/steamheater.py [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+        Define constraints which describe the behaviour of the unit model.
+        """
+        # Thermal conductivity of metal
+        self.therm_cond_metal = Param(
+                initialize=43.0,
+                mutable=True,
+                doc='Thermal conductivity of tube metal')
+        # Thermal conductivity of slag
+        self.therm_cond_slag = Param(
+                initialize=1.3,
```
Comment (andrewlee94) [inline idaes/power_generation/unit_models/steamheater.py:325]: For the future, but these would probably be better as `Vars` to allow for parameter estimation (slag conductivity strikes me as something we would want to estimate).

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A lot of comments but a lot are just observations that do not need to be fixed right now. There were a couple of typos I noticed and couple of places where we can do things better, but otherwise I think this is good.

**Review by andrewlee94 — DISMISSED** [review]

File: idaes/power_generation/unit_models/boiler_fireside.py [hunk 3]
```diff
@@ -43,7 +43,8 @@
 2. PA to raw coal flow rate ratio can be specified by a flowsheet constraint
 3. Fraction of coal moisture vaporized in mill should be consistent with
    the surrogate model (typically 0.6)
-4. Moisture mass fration in raw coal is a user input and should be consistent with the values used to train the surrogate model
+4. Moisture mass fration in raw coal is a user input and
```
Comment (andrewlee94) [inline idaes/power_generation/unit_models/boiler_fireside.py:46]: Typo: fraction.

File: idaes/power_generation/unit_models/boiler_fireside.py [hunk 4]
```diff
@@ -176,6 +177,12 @@ class BoilerFiresideData(UnitModelBlockData):
 
     def build(self):
         super(BoilerFiresideData, self).build()
+        # Insert user provided custom surrogate model function
+        # self.config.surrogate_function(self)
+        if self.config.surrogate_dictionary is None:
+            raise ConfigurationError('user needs to provide '
```
Comment (andrewlee94) [inline idaes/power_generation/unit_models/boiler_fireside.py:183]: Error messages really should start with a capital, and it is probably helpful to tell the user where the error occurred. I.e.

`raise ConfiguraitonError('{} Some message here'.format(self.name))`

This way, the user knows which model they need to fix.

**Review by andrewlee94 — COMMENTED** [review]
A couple of minor observations.

**Comment by eslickj** [comment]
Looks like there is a list in a doc string that doesn't end with a new line:

/home/runner/work/idaes-pse/idaes-pse/idaes/power_generation/unit_models/boiler_fireside.py:docstring of idaes.power_generation.unit_models.boiler_fireside:34: WARNING: Enumerated list ends without a blank line; unexpected unindent.

**Comment by andrewlee94** [comment]
I think the issue is on line 47 of boiler_fireside.py. Try not indenting the last line of the numbered list first.

**Comment by eslickj** [comment]
I think this looks good.  Once the tests pass, I'll approve.

**Review by jghouse88 — APPROVED** [review]
LGTM (Rubber stamping)

**Review by eslickj — APPROVED** [review]

**Pull request merged**